spec/openapi3.yaml

```diff
@@ -2969,6 +3203,83 @@ components:
       allOf:
         - $ref: "#/components/schemas/EmbeddedTransactionDTO"
         - $ref: "#/components/schemas/TransferTransactionBodyDTO"
+
+    # Schemas - Metadata
+    MetadataEntriesDTO:
+      type: object
+      required:
+        - metadataEntries
+      properties:
+        metadataEntries:
+          type: array
+          description: Array of account metadata.
+          items:
+            $ref: "#/components/schemas/KeyMetadataEntryDTO"
+    KeyMetadataEntryDTO:
+      type: object
+      required:
+      - key
+      - metadata
+      properties:
+        key:
+          type: integer
+          format: int64
+        metadata:
+          type: array
+          description: Array of metadata entry objects.
+          items:
+            $ref: "#/components/schemas/MetadataDTO"
+    MetadataDTO:
+      type: object
+      required:
+        - meta
+        - metadataEntry
+      properties:
+        meta:
+          $ref: "#/components/schemas/MetadataMetaDTO"
+        metadataEntry:
+          $ref: "#/components/schemas/MetadataEntryDTO"
+    MetadataMetaDTO:
+      type: object
+      required:
+        - id
+      properties:
+        id:
+          type: string
+    MetadataEntryDTO:
+      type: object
+      required:
+        - compositeHash
+        - senderPublicKey
+        - targetPublicKey
+        - scopedMetadataKey
+        - targetId
+        - metadataType
+        - valueSize
+        - value
+      properties:
+        compositeHash:
+          $ref: "#/components/schemas/Hash256"
+        senderPublicKey:
+          $ref: "#/components/schemas/PublicKey"
+        targetPublicKey:
+          $ref: "#/components/schemas/PublicKey"
+        scopedMetadataKey:
+          type: integer
+          format: int64
+        targetId:
```

Comment:
I believe that account metadata entries do not have the property targetId (please, double-check). If this is correct, and we don't want to create separate DTOs, what we can do is to define this property as optional (remove from ``required`` list) with type  ``anyof: MosaicId or NamespaceId``.

Reply:
`Anyof` doesn't really work on the generator anyway, so we gonna patch it anyway. So this one is not a show stopper, but without `targetId` looks better to me.

Reply:
unsnure about the property, thought seems logical, will remove the `required`